docs/arch/adr-011-kurl-addon-operator.md

```diff
@@ -0,0 +1,96 @@
+- It will install and confiure the KOTS admin console.
+- It can be installed on an existing [kURL](https://kurl.sh/) cluster with only Kubernetes, containerd and flannel.
+- The installation mechanism for add-ons will be through the use of Helm Charts.
+- By default the add-on configuration will be opinionated by us but we will enable the end-user to override the configuration for maximum flexibility. Vendors and users will be able to override the configuration using a standard [Helm values](https://helm.sh/docs/chart_template_guide/values_files/) file.
+- It will rely as much as possible on Helm for managing the lifecyle of the add-ons.
+
+### The `Installer` API
+
+The `Installer` API defines a resource for installing specific add-ons.
+
+```yaml
+apiVersion: cluster.kurl.sh/v2beta1
+kind: Installer
+metadata:
+  name: installer-sample
+spec:
+  channel: "1.27"
+  minio:
+    enabled: true
+    configOverride:
+        persistence:
+            enabled: true
+        replicas: 1
+        mode: standalone
+
+```
+
+### The Controllers
+
+The operator will have at minimum one controller which ensures the `Installer` CR is reconciled with the cluster periodically. The `Installer` controller (main controller) will delegate responsiblity for managing add-ons to other controllers. The reconcilation for the main controller will consist of determining the add-ons to be managed from the `Installer` CR and creating a controller for each add-on (addon-on controller).
+
+The add-on controller will be responsible for
+- verifying the requirements for the add-on are met
+- managing the add-on itself
+- ensure add-on is in desired state
+- update the add-on CR status
+
+### Managing the Add-ons
+
+How the add-ons will be managed is still an open question. We have a couple of options:
+
+_Use the Helm Controller provided by Flux_
+
+The [Flux Helm Controller](https://fluxcd.io/flux/components/helm/helmreleases/) affords us the following benefits:
+- Stable [HelmRelease API](https://fluxcd.io/flux/components/helm/api/)
+- Helm release [dependencies](https://fluxcd.io/flux/components/helm/helmreleases/#helmrelease-dependencies)
+- Support for Helm install, upgrade, test, rollback and uninstall
+- Performs Helm uninstall on `HelmRelease` CR removal
+- Chart values override\
+- Pull charts from Helm repository directly
+
+_Develop our own Homebrew Helm Controller_
+
+This approach affords us with a controller that meets our specific needs and avoids any third party software dependency. 
+
+
+
+
+## Status
+
+Proposed
+
+## Open Questions
+
+- How to allow full configuration of add-ons for advanced vendors/users?
+- Do we use an existing feature rich Helm controller like the one provided by [Flux](https://fluxcd.io/flux/components/helm/) or build our own?
+- How to structure the add-on to adhere to SOLID principles?
+
+## Consequences
+
+- A kURL CLI will be needed to install the operator dependencies (if any)
```

Comment:
To install the Operator we only need the k8s resources (which can be shipped in a single file and used via raw) and the Operator image. 

Therefore, besides be very cool and not so hard we have a CLI, we have a amazing prototype for that already is not be necessary to have a CLI to provide the Operator, see some options:

**We can do such as Prometheus Operator does**
- Provide all in a bundle/file: https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/release-0.51/bundle.yaml
- Then, to instal it we just need kubectl apply -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/release-0.51/bundle.yaml and then apply the kURL/Installer CR

Also, they allow download from the release page, see: https://github.com/prometheus-operator/prometheus-operator/releases/download/v0.65.1/bundle.yaml

- We can either have a small shell script to just ship the Operator
- We can either provide a tar with the instructions to untar and install